### PR TITLE
feat: Sqc 246 create modal UI using vanilla js without framework dependencies

### PR DESCRIPTION
**Pull request by amirsaran3**
# Description
Added new `modal-ui-js` package that does not use React for rendering components. Instead uses plain/vanilla JS.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

# Type of change.

- [ ] FIX - a PR of this type patches a bug.
- [x] FEATURE - a PR of this type introduces a new feature.
- [ ] BUILD - a PR of this type introduces build changes.
- [ ] CI - a PR of this type introduces CI changes.
- [ ] DOCS - a PR of this type introduces DOCS improvement.
- [ ] STYLE - a PR of this type introduces style changes.
- [ ] REFACTOR - a PR of this type introduces refactoring.
- [ ] PERFORMANCE - a PR of this type introduces performance changes.
- [ ] TEST - a PR of this type adds more tests.
- [ ] CHORE - a PR introduces other changes than the specified above.

### CODE REVIEW

File: packages/modal-ui-js/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# modal-ui-js
```
Comment (kujtimprenkuSQA) [inline packages/modal-ui-js/README.md:1]: We will need to update the Readme file if it's the same as the old modal just copy it here too.

File: packages/modal-ui-js/src/lib/render-modal.ts [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+}
+
+export function renderModal() {
+  if (!modalState) {
+    return;
+  }
+
+  modalState.container.innerHTML = `
+    <div class="nws-modal-wrapper">
+      <div class="nws-modal-overlay"></div>
```
Comment (kujtimprenkuSQA) [inline packages/modal-ui-js/src/lib/render-modal.ts:121]: Would it be possible to close the modal when we click on the overlay (outside the modal?

Also, it would be nice to close the modal when we press `ESC` like we have in the react modal.

File: tools/scripts/publish.mjs [hunk 3]
```diff
@@ -0,0 +1,61 @@
+/**
```
Comment (kujtimprenkuSQA) [inline tools/scripts/publish.mjs:1]: Is this file needed or was it added by NX automatically?

File: packages/modal-ui-js/src/lib/styles.css [hunk 4]
```diff
@@ -0,0 +1,1102 @@
...
+    !*** css ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[6].rules[0].oneOf[0].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[6].rules[0].oneOf[0].use[2]!./node_modules/resolve-url-loader/index.js??ruleSet[1].rules[6].rules[1].use[0]!./node_modules/@angular-devkit/build-angular/node_modules/sass-loader/dist/cjs.js??ruleSet[1].rules[6].rules[1].use[1]!./examples/angular/src/styles.scss (1) ***!
+    \*************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************/
+/* You can add global styles to this file, and also import other style files */
+#near-wallet-selector-modal {
```
Comment (kujtimprenkuSQA) [inline packages/modal-ui-js/src/lib/styles.css:10]: It looks like the modal is not responsive can you please align with the latest changes which are in the other branch mainly changes made by Damir:
https://github.com/near/wallet-selector/pull/422

File: packages/modal-ui-js/project.json [hunk 5]
```diff
@@ -0,0 +1,64 @@
+{
+  "$schema": "../../node_modules/nx/schemas/project-schema.json",
+  "sourceRoot": "packages/modal-ui-js/src",
+  "projectType": "library",
+  "targets": {
+    "build": {
+      "executor": "@nrwl/js:tsc",
```
Comment (kujtimprenkuSQA) [inline packages/modal-ui-js/project.json:7]: Would it be possible to change the builder to `rollup`?
Make it the same as in other wallets this way the bundle will be either in ESM or CJS otherwise, we can run into issues like we had before the package might not work on SSR frameworks.

File: packages/modal-ui-js/src/lib/components/WalletNotInstalled.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
+import { ModuleState, Wallet } from "@near-wallet-selector/core";
+import { modalState } from "../modal";
+
+export function renderWalletNotInstalled(module: ModuleState<Wallet>) {
+  if (!modalState) {
+    return;
+  }
+
+  document.querySelector(".modal-right")!.innerHTML = `
+    <div class="nws-modal-body">
```
Comment (kujtimprenkuSQA) [inline packages/modal-ui-js/src/lib/components/WalletNotInstalled.ts:10]: Can we please align this component and make it the same as the react modal in the redesign PR?

To see the difference please run the react example and check an Uninstalled injected wallet. Also, the blue "Open Wallet Name"  button should redirect to the extension.

File: packages/modal-ui-js/src/lib/render-modal.ts [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+  try {
+    if (module.type === "injected" && !module.metadata.available) {
+      return renderWalletNotInstalled(module);
+    }
+
+    if (module.metadata.deprecated) {
+      return renderWalletConnectionFailed(
+        module,
+        new Error("Wallet is depredacted")
```
Comment (kujtimprenkuSQA) [inline packages/modal-ui-js/src/lib/render-modal.ts:81]: There's a typo here the word `depredacted` should be `deprecated`

File: packages/modal-ui-js/src/lib/modal.ts [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+
+export let modalState: ModalState | null = null;
+
+export function updateModalState(newModalState: ModalState) {
+  modalState = newModalState;
+}
+
+const el = document.createElement("div");
+el.id = MODAL_ELEMENT_ID;
+document.body.appendChild(el);
```
Comment (kujtimprenkuSQA) [inline packages/modal-ui-js/src/lib/modal.ts:37]: We will need to wrap these three lines in an if statement to check if the window is defined otherwise it will fail in SSR frameworks. 

```ts
if (typeof window !== "undefined") {
  const el = document.createElement("div");
  el.id = MODAL_ELEMENT_ID;
  document.body.appendChild(el);
}
```

File: packages/modal-ui-js/package.json [hunk 9]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "@near-wallet-selector/modal-ui-js",
+  "version": "0.0.1",
+  "type": "commonjs"
```
Comment (kujtimprenkuSQA) [inline packages/modal-ui-js/package.json:4]: Can you make update the version and make it to the current one `6.0.1`?
Also, you will need to remove the `type` property.

**Review by HarisSQA — APPROVED** [review]

File: packages/modal-ui-js/src/lib/components/GetAWallet.ts [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+      "beforeend",
+      `
+    <div class="single-wallet-get">
+      <div class="icon">
+      <img
+          src="${filteredModules[i].metadata.iconUrl}"
+          alt="${filteredModules[i].metadata.name}"></div>
+      <div class="content">
+        <div class="title">${filteredModules[i].metadata.name}</div>
+        <div class="description">${
```
Comment (HarisSQA) [inline packages/modal-ui-js/src/lib/components/GetAWallet.ts:86]: If there isn't metadata description on field we would need to add custom text

**Review by HarisSQA — APPROVED** [review]
Good job. Just description in wallets if not defined, should we define it to have something or?

**Review by kujtimprenkuSQA — DISMISSED** [review]

**Review by AmmarHumackicSQA — DISMISSED** [review]

**Review by AmmarHumackicSQA — APPROVED** [review]
